core/templates/components/forms/schema-based-editors/schema-based-float-editor.component.ts

```diff
@@ -247,17 +247,18 @@ export class SchemaBasedFloatEditorComponent
       this.errorStringI18nKey = null;
     } else {
       // Make sure number is in a correct format.
+      let currentDecimalSeparator = this.getCurrentDecimalSeparator();
       let error = this.numericInputValidationService.validateNumericString(
-        this.localStringValue,
-        this.getCurrentDecimalSeparator()
+        this.localStringValue.replace(/\.|,/g, currentDecimalSeparator),
+        currentDecimalSeparator
       );
       if (error !== undefined) {
         this.localValue = null;
         this.errorStringI18nKey = error || null;
       } else {
         // Parse number if the string is in proper format.
         this.localValue = this.numberConversionService.convertToEnglishDecimal(
-          this.localStringValue
+          this.localStringValue.replace(/\.|,/g, currentDecimalSeparator)
```

Comment:
A question: Instead of doing localStringValue.replace() in multiple places, should we just be storing the right thing in there from the outset?

Reply:
Thank you for this!
I have improved it in the commit i am pushing now